core/src/main/java/com/sequenceiq/cloudbreak/service/multiaz/MultiAzCalculatorService.java

```diff
@@ -36,9 +41,11 @@
 @Service
 public class MultiAzCalculatorService {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(MultiAzCalculatorService.class);
+
     private static final String DEFAULT_RACK = "default-rack";
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(MultiAzCalculatorService.class);
+    private static final int MINIMUM_NUMBER_OF_HINTS = 3;
```

Comment:
This should be configurable for 2 reasons  :  

1. number of hints can change based on number of available AZs by cloud provider so we should not make code change here.
2. This can also serve the purpose of disabling this feature by setting this configuration to a high value.